### DISCUSSION TITLE
let Object.get_class() work with custom types

### DISCUSSION THREAD

**Discussion by azur-wolve** [godotengine/godot-proposals#5173] [Engine Core]
Currently can't use `Object.get_class()` for custom types:
```
String get_class() const

    Returns the object's class as a String. See also is_class().

    Note: get_class() does not take class_name declarations into account.
    If the object has a class_name defined, the base class name will be returned instead.
```
What about have a virtual method `Object._get_class()` to override and customize it?
Or even better, just let Godot return the name of the custom type as string.

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/21789.
